src/Domain/CoordinateMaps/TimeDependent/Translation.cpp

```diff
@@ -84,25 +105,73 @@ std::optional<std::array<double, Dim>> Translation<Dim>::inverse(
              << function_of_time.size()
              << ") does not match the dimension of the translation map (" << Dim
              << ").");
-  double radial_function_value = 1.0;
-  if (f_of_r_ != nullptr) {
-    const double root = root_finder(target_coords - center_, function_of_time);
-    radial_function_value = (*f_of_r_)(root);
-  }
-  for (size_t i = 0; i < Dim; i++) {
-    gsl::at(result, i) -= function_of_time[i] * radial_function_value;
+  if (std::isnan(inner_radius_)) {
+    double radial_function_value = 1.0;
+    if (f_of_r_ != nullptr) {
+      const double root =
+          root_finder(target_coords - center_, function_of_time);
+      radial_function_value = (*f_of_r_)(root);
+    }
+    for (size_t i = 0; i < Dim; i++) {
+      gsl::at(result, i) -= function_of_time[i] * radial_function_value;
+    }
+    return result;
+  } else {
+    auto target_radius = magnitude(target_coords);
+    double non_translated_radius = 0.;
+    for (size_t i = 0; i < Dim; i++) {
+      non_translated_radius +=
+          square(gsl::at(target_coords, i) - function_of_time[i]);
+    }
+    non_translated_radius = sqrt(non_translated_radius);
+    if (target_radius >= outer_radius_) {
+      return result;
+    } else if (non_translated_radius <= inner_radius_) {
+      for (size_t i = 0; i < Dim; i++) {
+        gsl::at(result, i) -= function_of_time[i];
+      }
+    } else {
+      // Solving a quadratic of known quantities.
```

Comment:
Can you add more to this code comment explaining which exact quadratic equation you are solving and what `a`, `b`, `c` are explicitly? That'll go a long way in the future when somebody else has to look at this (like you are in SpEC right now!).